compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCastGen.java

```diff
@@ -133,12 +137,13 @@
 
     //the symbol table is currently set from package gen class
     private final SymbolTable symbolTable;
-
     private final JvmTypeGen jvmTypeGen;
+    private final Types types;
 
-    public JvmCastGen(SymbolTable symbolTable, JvmTypeGen jvmTypeGen) {
+    public JvmCastGen(SymbolTable symbolTable, JvmTypeGen jvmTypeGen, CompilerContext compilerContext) {
         this.symbolTable = symbolTable;
         this.jvmTypeGen = jvmTypeGen;
+        this.types = Types.getInstance(compilerContext);
```

Comment:
Shall we pass the types intead of passing compilerContext here. like symbolTable